libkineto/include/ActivityType.h

```diff
@@ -40,12 +41,25 @@ enum class ActivityType {
     XPU_RUNTIME, // host side xpu runtime events
     COLLECTIVE_COMM, // collective communication
 
+    // PrivateUse1 Activity types
+    PRIVATEUSE1_USER_ANNOTATION,
+    PRIVATEUSE1_MEMCPY,
+    PRIVATEUSE1_MEMSET,
+    PRIVATEUSE1_CONCURRENT_KERNEL, // privateUse1 on-device kernels
+    PRIVATEUSE1_EXTERNAL_CORRELATION,
+    PRIVATEUSE1_RUNTIME, // host side privateUse1 runtime events
+    PRIVATEUSE1_DRIVER, // host side privateUse1 driver events
+    PRIVATEUSE1_SYNC, // synchronization events between runtime and kernels
```

Comment:
Could we re-use the above ActivityTypes for most of these, and add PRIVATEUSE1_RUNTIME and PRIVATEUSE1_DRIVER only like MTIA above?

Reply:
We need to get the DeviceType::PrivateUse1 from Activities. So, i think these five ActivityTypes https://github.com/pytorch/pytorch/pull/120556/files#diff-11a30835eab7801f23620d212a8b7347c87530432971d8777a92f937d31a0bb9R354-R359 are still needed.

Of course, like EXTERNAL_CORRELATION,  can indeed be reused. I will delete PRIVATEUSE1_EXTERNAL_CORRELATION.

Reply:
I noticed that XPU uses another way to get DeviceType. And [pr](https://github.com/pytorch/pytorch/pull/120185/files#diff-11a30835eab7801f23620d212a8b7347c87530432971d8777a92f937d31a0bb9R327) had been merged. If so, We can indeed reuse most of the these, and only need to add PRIVATEUSE1_RUNTIME and PRIVATEUSE1_DRIVER.

Reply:
cc @xunsongh, would be great to get cross-review since XPU and PrivateUse1 may share very similar implementations.

Reply:
I have no objections to this section, and I appreciate your efforts in reusing the code while attempting to maintain consistency with the XPU modifications. Thank you.